### PR TITLE
[docs] configuration.md - grammar & word-smithing

### PR DESCRIPTION
**Pull request by gregfenton**
# Why

Overly worded and hard to parse.

# How

Docs fixes.

# Test Plan

(docs review)

### CODE REVIEW

File: docs/pages/workflow/configuration.md [hunk 1]
```diff
@@ -103,8 +103,8 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
...
-4. Whatever is returned from the dynamic config will be used as the final config. It cannot have any promises.
-5. All functions in the config will be evaluated and serialized before any tool in the Expo ecosystem uses it. This is because the config must be a json manifest when it's hosted.
+1. First the static config will be read if `app.config.json` exists, then falling back to `app.json`. If no static config exists then default values are inferred from the `package.json` and installed node modules.
+2. The dynamic config will read if `app.config.ts` or `app.config.js` exist. If both exist then the TypeScript config is used.
+3. If the dynamic config returns a function, then the static config is passed to the function with `({ config }) => ({})`. This function can mutate the static config values.
+4. Whatever is returned from the dynamic config is used as the final config. It cannot have any promises.
+5. All functions in the config are evaluated and serialized before any tool in the Expo ecosystem uses it. The config must be a JSON manifest when it is hosted.
```
Comment (cruzach) [inline docs/pages/workflow/configuration.md:110]:
```suggestion
1. The static config will be read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.
2. The dynamic config will be read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.
3. If the dynamic config returns a function, then the static config is passed to the function with `({ config }) => ({})`. This function can then mutate the static config values.
4. Whatever is returned from the dynamic config is used as the final config. It cannot have any promises.
5. All functions in the config are evaluated and serialized before any tool in the Expo ecosystem uses it. The config must be a JSON manifest when it is hosted.
```

File: docs/pages/workflow/configuration.md [hunk 2]
```diff
@@ -103,8 +103,8 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
 
 There are two different types of configs static (`app.config.json`, `app.json`) and dynamic (`app.config.js`, `app.config.ts`). Static configs can be automatically updated with CLI tools, whereas dynamic configs must be manually updated by the developer.
```
Comment (cruzach) [inline docs/pages/workflow/configuration.md:104]:
```suggestion
There are two different types of configs: static (`app.config.json`, `app.json`), and dynamic (`app.config.js`, `app.config.ts`). Static configs can be automatically updated with CLI tools, whereas dynamic configs must be manually updated by the developer.
```

**Review by cruzach — CHANGES_REQUESTED** [review]
thanks for fixing this up! Just a few wording changes from me

**Review by gregfenton — COMMENTED** [review]
Reword to be explicit and clear - remove conditional wording

File: docs/pages/workflow/configuration.md [hunk 3]
```diff
@@ -101,10 +101,10 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
...
 
-1. First the static config will be read by checking first if `app.config.json` exists, then falling back on app.json. If no static config exists then default values will be inferred from the package.json and installed node modules.
-2. The dynamic config will read by first checking if `app.config.ts` or `app.config.js` exists. If both exist then the TypeScript config will be used.
-3. If the dynamic config returns a function, then the static config will be passed to the function with `({ config }) => ({})`. This can be used to mutate the static config values.
-4. Whatever is returned from the dynamic config will be used as the final config. It cannot have any promises.
-5. All functions in the config will be evaluated and serialized before any tool in the Expo ecosystem uses it. This is because the config must be a json manifest when it's hosted.
+1. The static config are read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.
```
Comment (cruzach) [inline docs/pages/workflow/configuration.md:106]:
```suggestion
1. The static config is read if `app.config.json` exists (falls back to `app.json`). If no static config exists, then default values are inferred from the `package.json` and your dependencies.
```

File: docs/pages/workflow/configuration.md [hunk 4]
```diff
@@ -101,10 +101,10 @@ export default ({ config }: ConfigContext): ExpoConfig => ({
...
+2. The dynamic config are read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.
```
Comment (cruzach) [inline docs/pages/workflow/configuration.md:107]:
```suggestion
2. The dynamic config is read if either `app.config.ts` or `app.config.js` exist. If both exist, then the TypeScript config is used.
```

**Review by cruzach — APPROVED** [review]
